include/fiction/algorithms/physical_design/design_sidb_gates.hpp

```diff
@@ -366,11 +384,18 @@ class design_sidb_gates_impl
      *
      * @return A vector of designed SiDB gate layouts.
      */
-    [[nodiscard]] std::vector<Lyt> run_quickcell() const noexcept
+    [[nodiscard]] std::vector<Lyt> run_quickcell() noexcept
     {
         mockturtle::stopwatch stop{stats.time_total};
         const auto            gate_candidates = run_pruning();
 
+        stats.number_of_layouts_after_first_pruning =
+            all_canvas_layouts.size() - number_of_discarded_layouts_at_first_pruning;
+        stats.number_of_layouts_after_second_pruning =
+            stats.number_of_layouts_after_first_pruning - number_of_discarded_layouts_at_second_pruning;
+        stats.number_of_layouts_after_third_pruning =
+            stats.number_of_layouts_after_second_pruning - number_of_discarded_layouts_at_third_pruning;
```

Comment:
To atomically obtain the values stored in `std::atomic`, you should use `.load()`.

https://en.cppreference.com/w/cpp/atomic/atomic

Reply:
oh, you are right! Many thanks!